newIDE/app/src/CommunityLeaderboards/CommunityLeaderboardsContext.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import * as React from 'react';
+import {
+  getUserCommentQualityRatingsLeaderboards,
+  type UserLeaderboard,
+} from '../Utils/GDevelopServices/User';
+import {
+  getGameCommentQualityRatingsLeaderboards,
+  type GameLeaderboard,
+} from '../Utils/GDevelopServices/Game';
+
+type CommunityLeaderboardsState = {|
+  userLeaderboards: Array<UserLeaderboard> | null,
+  gameLeaderboards: Array<GameLeaderboard> | null,
+  error: ?Error,
+  fetchCommunityLeaderboards: () => void,
+|};
+
+export const CommunityLeaderboardsContext = React.createContext<CommunityLeaderboardsState>(
+  {
+    userLeaderboards: null,
+    gameLeaderboards: null,
+    error: null,
+    fetchCommunityLeaderboards: () => {},
+  }
+);
+
+type CommunityLeaderboardsStateProviderProps = {|
+  children: React.Node,
+|};
+
+export const CommunityLeaderboardsStateProvider = ({
+  children,
+}: CommunityLeaderboardsStateProviderProps) => {
+  const [userLeaderboards, setUserLeaderboards] = React.useState<
+    Array<UserLeaderboard>
+  >([]);
+  const [gameLeaderboards, setGameLeaderboards] = React.useState<
+    Array<GameLeaderboard>
+  >([]);
+
+  const [error, setError] = React.useState<?Error>(null);
+  const isLoading = React.useRef<boolean>(false);
+
+  const fetchCommunityLeaderboards = React.useCallback(() => {
+    if (isLoading.current) return;
+
+    (async () => {
```

Comment:
You don't need to do that, you can use an async callback

Reply:
Good catch, I copy pasted this from another context.